presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -370,6 +369,22 @@ private PlanBuilder explicitCoercionSymbols(PlanBuilder subPlan, Iterable<Symbol
                 analysis.getParameters());
     }
 
+    private List<Expression> constructProjectExpressions(QuerySpecification node)
+    {
+        List<Expression> arguments = analysis.getAggregates(node).stream()
+                .map(FunctionCall::getArguments)
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+        // filter expressions need to be projected first
+        List<Expression> filterExpressions = analysis.getAggregates(node).stream()
+                .map(FunctionCall::getFilter)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+        arguments.addAll(filterExpressions);
```

Comment:
This is incorrect. There's no guarantee the result of `Collectors.toList()` is mutable.

You can create an `ImmutableList.Builder` above and do `.forEach(arguments::addAll)` instead of `.collect()`.